pyproject.toml

```diff
@@ -41,8 +35,7 @@ requires = [
     "numpy==1.22.3; python_version=='3.10' and platform_system=='Windows' and platform_python_implementation != 'PyPy'",
 
     # default numpy requirements
-    "numpy==1.19.5; python_version=='3.8' and (platform_machine!='arm64' or platform_system!='Darwin') and platform_machine!='aarch64' and platform_machine!='loongarch64' and platform_python_implementation != 'PyPy'",
-    "numpy==1.19.5; python_version=='3.9' and (platform_machine!='arm64' or platform_system!='Darwin') and platform_machine!='loongarch64' and platform_python_implementation != 'PyPy'",
+    "numpy==1.21.6; python_version=='3.9' and (platform_machine!='arm64' or platform_system!='Darwin') and platform_machine!='loongarch64' and platform_python_implementation != 'PyPy'",
```

Comment:
Still doesn't look 100% right. I didn't test, but I am fairly sure that `and (platform_machine!='arm64' or platform_system!='Darwin')` should now be removed here, because the alternative condition higher up was dropped. You need one, and exactly one, condition for every platform/OS/interpreter combo.

Reply:
```suggestion
    "numpy==1.21.6; python_version=='3.9' and platform_machine!='loongarch64' and platform_python_implementation != 'PyPy'",
```